### ISSUE TITLE
Restrict props to `Record`

### ISSUE THREAD

**Issue by natefaubion** [purescript-contrib/purescript-react#119]
You are guaranteed buggy behavior otherwise.

**Comment by coot** [comment]
Indeed, we could wrap props in a record like it is dome with the state already.  Just limiting to records will rule out newtype wrappers of records (or we could type check this with a type class).

**Comment by coot** [comment]
We could wait for instance chains in `11.6.0` and use something as simple as this:
```
class IsRecord a

instance recordIsRecord :: IsRecord {|r}
else instance newtypeRecordIsRecord :: Newtype a {|r} => IsRecord a
```

**Comment by ethul** [comment]
Closed by #129
